### PR TITLE
fix AUTO wrong device selection logic if dynamic stateful model

### PR DESCRIPTION
**Pull request by songbell**
### Details:
 - check dynamic + stateful logic

### Tickets:
 - CVS-140254, CVS-140844

### CODE REVIEW

File: src/plugins/auto/src/plugin.cpp [hunk 1]
```diff
@@ -837,10 +837,9 @@ std::vector<DeviceInformation> Plugin::filter_device_by_model(const std::vector<
     // If CPU is in candidate list, load dynamic model to CPU first
     // For MULTI do not only load stateful model to CPU
     // For AUTO CTPUT only load stateful model to CPU
```
Comment (yangwang201911) [inline src/plugins/auto/src/plugin.cpp:839]: should update comment here

File: src/plugins/auto/src/plugin.cpp [hunk 2]
```diff
@@ -852,25 +846,35 @@ std::vector<DeviceInformation> Plugin::filter_device_by_model(const std::vector<
...
+        return filter_device;
     }
 
     // disable CPU_HELP and runtime fallback if model is stateful
     disable_startup_runtime_fallback();
 
     auto is_supported_stateful = [&](const std::string& device_name, const ov::AnyMap& config) {
-        auto device_qm = get_core()->query_model(model, device_name, config);
+        auto query_config = config;
+        // remove core property from config, to be checked further if can do this elegantly
```
Comment (songbell) [inline src/plugins/auto/src/plugin.cpp:857]: @ilya-lavrenov @riverlijunjie meta plugin may need to query model for next steps(auto/hetero), but the passed property may contain core level properties, what do you suggest in this case? in general we don't touch these properties in meta plugin, but hw plugin query model will throw exception if been queries with cache_dir, batching properties etc..

Comment (ilya-lavrenov) [reply src/plugins/auto/src/plugin.cpp:857]: why cache_dir is passed to query_model? 
If user do it, it's his mistake

Comment (songbell) [reply src/plugins/auto/src/plugin.cpp:857]: @ilya-lavrenov  user just pass the cache_dir or batch property to compile model, but inside AUTO or hetero, for example, AUTO will call query model to decide on the final candidate list, hetero will call query model to decide on the graph splitting logic, so:
1) we remove the unsupported properties inside plugin, but in this way we will have to touch these properties again in plugin , which I think is not  reasonable under current design logic
2) query model to remove only core properties
3) before query model, we add extra call inside meta plugin to filter only plugin supported properties, similar to current get_supported_properties API, but do not return core properties?

File: src/plugins/auto/tests/unit/stateful_model_test.cpp [hunk 3]
```diff
@@ -249,19 +277,78 @@ const std::vector<StatefulModelConfigParams> testConfigs = {
...
+                              true,
+                              std::map<std::string, bool>{{"GPU", true}},
+                              true,
+                              std::vector<std::pair<std::string, int>>{{"GPU", 1}}},
+    StatefulModelConfigParams{"GPU",
+                              true,
+                              true,
+                              std::map<std::string, bool>{{"GPU", false}},
+                              true,
+                              std::vector<std::pair<std::string, int>>{{"GPU", 1}}},
```
Comment (songbell) [inline src/plugins/auto/tests/unit/stateful_model_test.cpp:297]: does the test check that enable start up fall back is disabled when CPU and GPU both support stateful?

File: src/plugins/auto/src/plugin.cpp (hunk 2 above)
Comment (songbell) [reply src/plugins/auto/src/plugin.cpp:857]: let's try avoid calling query model from compiled model, it's expensive especially for large models @yangwang201911

Comment (yangwang201911) [reply src/plugins/auto/src/plugin.cpp]: Updated.

File: src/plugins/auto/tests/unit/stateful_model_test.cpp (hunk 3 above)
Comment (yangwang201911) [reply src/plugins/auto/tests/unit/stateful_model_test.cpp]: Updated.

**Comment by songbell** [comment]
@yangwang201911 need to make CI green

**Review by peterchen-intel — APPROVED** [review]

**Review by zhaixuejun1993 — APPROVED** [review]

**Review by yangwang201911 — APPROVED** [review]

**Comment by songbell** [comment]
@wangleis please help merge

**Review by wangleis — APPROVED** [review]

**Pull request merged**